.test-infra/tools/python_installer.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+#
+#    Licensed to the Apache Software Foundation (ASF) under one or more
+#    contributor license agreements.  See the NOTICE file distributed with
+#    this work for additional information regarding copyright ownership.
+#    The ASF licenses this file to You under the Apache License, Version 2.0
+#    (the "License"); you may not use this file except in compliance with
+#    the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the spefic language governing permissions and
+#    limitations under the License.
+#
+#    Python version 3.6.13-3.7.10-3.8.9-3.9.4 installer via pyenv.
+#
+set -euo pipefail
+
+# Variable containing the python versions to install
+python_versions_arr=("3.6.13" "3.7.10" "3.8.9" "3.9.4")
```

Comment:
```suggestion
python_versions_arr=("3.8.9" "3.6.13" "3.7.10" "3.9.4" "3.10.1")
```

Explanation: 
1) the first version in the list will be the default alias for python, python3 commands. We can also add a comment about it. Python 3.8 seems like a good choice atm since Python 3.6 is already EOL and Python 3.9 is not yet supported by Beam
2) Let's add Python 3.10.1 - looks like it's already supported by pyenv. this may be helpful in later this year when we start working on Python 3.10 support. Might as well add it now since we are rebuilding the image anyway.